src/tests/JIT/opt/OSR/synchronized.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+struct S
+{
+    public long y;
+    public int x;
+}
+
+class Z
+{
+    virtual public S F()
+    {
+        S s = new S();
+        s.x = 100;
+        s.y = -1;
+        return s;
+    }
+}
+
+class X
+{
+    Z z;
+
+    [MethodImpl(MethodImplOptions.NoInlining | MethodImplOptions.Synchronized)]
+    public S G()
+    {
+        S s = new S();
+
+        for (int i = 0; i < 100_000; i++)
+        {
+            s = z.F();
```

Comment:
Does it make sense to verify `Monitor.IsEntered(this)` inside the loop?

Reply:
Sure, will add this.